docs/user/advanced.rst

```diff
@@ -482,8 +482,15 @@ To use HTTP Basic Auth with your proxy, use the `http://user:password@host/` syn
         "http": "http://user:pass@10.10.1.10:3128/",
     }
 
-Note that proxy URLs must include the scheme.
+To give a proxy for a specific scheme and host, use the
+`scheme://host` form for the key::
 
+    proxies = {
+      "http://10.20.1.128": "http://10.10.1.10:5323",
+    }
+
+Note that proxy URLs must include the scheme.
+    
```

Comment:
Can we comment here that this is *not* a longest-prefix-match, but it must match *exactly*? E.g. If you set `proxies={'http://10.21.1.128': 'some_url'}`, you will not get that proxy for `http://10.21.1.128:8080`. I feel like this could otherwise trip people up.